### PR TITLE
fix: missing Oci-Subject header pushing index with subject

### PR DESCRIPTION
**Pull request by jdolitsky**
Also, run OCI conformance tests from HEAD (vs. 1.0.0)

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rchincha** [comment]
@jdolitsky some sanity checks in CI

Also pls sign your commit.

```
Run gsactions/commit-message-checker@v2
Checking commit messages against "^((build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test)(\(.+\))?(!)?(: (.*\s*)*))"...
- failed: "Fix missing Oci-Subject header when pushing index with subject

Also, run OCI conformance tests from HEAD (vs. 1.0.0)

Signed-off-by: Josh Dolitsky <josh@dolit.ski>"
Error: Error: Your first line has to the Conventional Commits specification.
```

File: .github/workflows/oci-conformance-action.yml [hunk 1]
```diff
@@ -34,17 +34,29 @@ jobs:
...
-    - name: Run OCI Distribution Spec conformance tests
-      uses: opencontainers/distribution-spec@main
+    - name: Checkout OCI distribution-spec at HEAD
+      uses: actions/checkout@v3
+      with:
+        path: distribution-spec
+        # TODO: change to upstream once the foloowing PR is merged:
+        # https://github.com/opencontainers/distribution-spec/pull/436
+        repository: sudo-bmitch/distribution-spec
+        ref: pr-conformance-index-subject
```
Comment (jdolitsky) [inline .github/workflows/oci-conformance-action.yml:44]: FYI

**Comment by jdolitsky** [comment]
related: https://github.com/opencontainers/distribution-spec/pull/436

**Review by andaaron — APPROVED** [review]
Merge this as-is or wait on https://github.com/opencontainers/distribution-spec/pull/436 to be merged first and change the branch?

**Comment by jdolitsky** [comment]
Unfortunately https://github.com/opencontainers/distribution-spec/pull/436 will not pass CI/CD until there is a new zot release available (zot container image). My vote would be to get this change into an RC, then we update the OCI PR to point to new image cc @sudo-bmitch

**Comment by jdolitsky** [comment]
I can also drop the changes here related to conformance and you all can add them back later. Just wanted a way to test this change

**Comment by jdolitsky** [comment]
> I can also drop the changes here related to conformance and you all can add them back later. Just wanted a way to test this change

I did this in https://github.com/project-zot/zot/pull/1589 just in case

File: pkg/storage/s3/s3.go [hunk 2]
```diff
@@ -418,6 +418,19 @@ func (is *ObjectStorage) PutImageManifest(repo, reference, mediaType string, //n
...
+	} else if mediaType == ispec.MediaTypeImageIndex {
+		var index ispec.Index
+
+		err := json.Unmarshal(body, &index)
+		if err != nil {
+			return "", "", err
+		}
+
+		if index.Subject != nil {
+			subjectDigest = index.Subject.Digest
```
Comment (andaaron) [inline pkg/storage/s3/s3.go:430]: I'm working on a test to cover this.

**Comment by andaaron** [comment]
I updated https://github.com/project-zot/zot/pull/1589/files. This PR would conflict with https://github.com/project-zot/zot/pull/1559 on enabling the conformance code.

**Comment by jdolitsky** [comment]
@andaaron - feel free to close this PR in favor of #1589

**Comment by andaaron** [comment]
Closed in favor of https://github.com/project-zot/zot/pull/1589

**Pull request closed without merging**